bokeh/util/notebook.py

```diff
@@ -31,13 +34,21 @@ def load_notebook(resources=None, verbose=False, hide_banner=False, load_timeout
 
     '''
     html, js = _load_notebook_html(resources, verbose, hide_banner, load_timeout)
-    publish_display_data({'text/html': html})
-    publish_display_data({'application/javascript': js})
+    if notebook_type=='jupyter':
+        publish_display_data({'text/html': html})
+        publish_display_data({'application/javascript': js})
+    else:
+        _publish_zeppelin_data(html, js)
+
 
 FINALIZE_JS = """
 document.getElementById("%s").textContent = "BokehJS is loading...";
 """
 
+def _publish_zeppelin_data(html, js):
+    print('%html ' + html)
+    print('%html ' + '<script type="text/javascript">' + js + "</script>")
+
```

Comment:
Thanks @canavandl for review, zeppelin's python interpreter (JVM) would capture python process's stdout and send it to frontend. So yes, this is the standard way for zeppelin to output things to frontend.